keps/sig-security/3203-auto-refreshing-official-cve-feed/README.md

```diff
@@ -0,0 +1,690 @@
+- [x] Label those issues with `official-cve-feed` using https://docs.github.com/en/rest/reference/issues REST API
+- [x] https://github.com/kubernetes/committee-security-response/pull/133
+
+
+### Overview
+
+- Generate a JSON document using the results from the filtered label on `k/k` repo.
+- Create a Prow job to periodically generate this JSON document.
+- Update the JSON doc when needed (e.g. when a new CVE is announced) in `k/website`
+- Using Hugo, publish the list from this JSON document on official k8s website
+
+### Risks and Mitigations
+
+#### JSON blob construction will fail
+
+If this happens, we expect the job too fail. If blob construction fails, the failure will alert the owners of this feature and we will take action as needed. If the failure can not be fixed in a reasonable amount of time, the CVE feed will be stale until it is fixed. In case of urgency from community to get the refreshed feed, JSON blob will be manually updated via usual PR review and approval process.
+
+#### Urgent CVE feed refresh
+
+In some extenuating circumstances, we may need to update the CVE feed within minutes of the official CVE announcement, instead of waiting for the merge based or periodical website rebuild. In those situations, manual updates to JSON blob using usual PR reviews and approval process can be implemented.
+
+#### Misuse of Auto-Refresh feature
+
+Without proper filtering and control over who can label GitHub issues, the list of CVEs can become a list with poor signal to noise ratio making the list unusable. 
+
+For this purpose, the filtering is applied such that only issues that are marked as `closed`
+will be part of the list. Also, additionally, the `official-cve-feed` label is a 
+[restricted](https://github.com/kubernetes/test-infra/blob/master/config/prow/plugins.yaml#L140-L150)
+label that can only be applied by SRC and SIG Security Tooling Leads.
+
+#### Storage of CVE feed blob
+
+There are two options to store the CVE feed JSON blob:
+* __Google Cloud Bucket__: A new google cloud bucket can be created where the CVE feed is written and read using `gsutil` tool.
+* __Git repository__: Store it as a version controlled artifact in one of the kubernetes org websites.
+
+Google cloud option has an advantage of transparent updates to JSON blob where the prow job run will be identical everytime. The disadvantage with Google Cloud bucket is that, it will have an unofficial looking URL which would be hard for an end user to deciper it's authenticity and provenance.
+
+Advantage with git repository especially `k/website` hosts the JSON blob the domain name in the URL would be `k8s.io/static/security/official-cve-feed.json` which is much more recognizable, intuitive in terms of trust, TLS enabled and unlikely to be spoofed. The disadvantage though is that this might get delayed by PR review and approval process. However, this can be prevented through use of `skip-review` label.
```

Comment:
I'd expect to see [the choice we didn't select] mentioned under the heading “Alternatives”, later in the KEP.

Reply:
Refactored under alternatives